internal/app/machined/pkg/runtime/v1alpha1/platform/metal/metal.go

```diff
@@ -51,7 +51,7 @@ func (m *Metal) Configuration(ctx context.Context, r state.State) ([]byte, error
 	getURL := func() string {
 		downloadEndpoint, err := PopulateURLParameters(ctx, *option, r)
 		if err != nil {
-			log.Fatalf("failed to populate talos.config fetch URL: %q ; %s", *option, err.Error())
+			log.Printf("failed to populate talos.config fetch URL: %q ; %s", *option, err.Error())
```

Comment:
would this retry?

Reply:
yes, but in non-obvious way - this function gets called from `download.Download` on each attempt, so the error is not fatal - next time if will be called if the download fails from partially expanded URL.